geode-core/src/test/java/org/apache/geode/management/internal/api/LocatorClusterManagementServiceTest.java

```diff
@@ -125,16 +127,41 @@ public void before() throws Exception {
     doReturn(true).when(persistenceService).lockSharedConfiguration();
     doNothing().when(persistenceService).unlockSharedConfiguration();
     operationManager = mock(OperationManager.class);
+    dLockService = mock(DistributedLockService.class);
+
     service =
         spy(new LocatorClusterManagementService(cache, persistenceService, managers, validators,
             memberValidator, cacheElementValidator, operationManager));
+    doReturn(dLockService).when(service).getCmsDlockService();
 
     regionConfig = new Region();
     regionConfig.setName("region1");
 
     rebalanceOperation = new RebalanceOperation();
   }
 
+  @Test
+  public void lockAndUnlockCalledAtCreate() {
+    try {
+      service.create(regionConfig);
+    } catch (Exception ignore) {
+    }
+
+    verify(dLockService).lock(LocatorClusterManagementService.CMS_DLOCK_SERVICE_NAME, -1, -1);
+    verify(dLockService).unlock(LocatorClusterManagementService.CMS_DLOCK_SERVICE_NAME);
+  }
+
+  @Test
+  public void lockAndUnlockCalledAtDelete() {
+    try {
+      service.delete(regionConfig);
+    } catch (Exception ignore) {
```

Comment:
Can we remove try-block and use `throws Exception`?

Reply:
in this test setup, a NPE will already be thrown. This test is to make sure the two calls always get called when any exception happens. I changed the test method to be `lockAndUnlockCalledAtDeleteWithException`. And added the verification to another test setup when deletion is successful.